### PR TITLE
Doc 458 explain cursors

### PR DESCRIPTION
**Pull request by ghost**
- Added a more detailed explanation of what the cursor is
- Corrected code samples (Fist example was not working, the second example change is just syntax sugar to be aligned with the previous)

### CODE REVIEW

**Review by diana-macrometa — APPROVED** [review]
This is a good start. I think we also need a topic in the Queries and Query Workers section explaining what a cursor is. That topic could then link to this and to CLI commands, API commands, and so on.

I suggest a separate topic for that one.

File: docs/collections/returning-collection-data.md
Comment (jamesm-macrometa) [inline docs/collections/returning-collection-data.md]:
```suggestion
We can create a cursor to receive the results from our request. A cursor details where we left off when reading a stream of responses from API, so we can continue to read the batch of required data. When calling the `query` method request options are provided as key/value pairs. For a detailed explanation of options visit the [API reference](https://macrometa.com/docs/api#/operations/createQueryCursor). Set the `batchSize` to the required value (code ex. 1000) and the `stream` value to true. This will permit the results from the query to be returned as a stream, allowing us to return subsequent batches of data. In response, the entire result set is loaded into the cursor (code ex. `cursor`). Therefore we must be mindful of client-side memory capacity when running queries that can potentially return a large result set.
```

A few minor grammatical/concision suggestions here.  Also, no need to say "please" in technical docs.

Otherwise looks good!

**Review by jamesm-macrometa — APPROVED** [review]

File: docs/collections/returning-collection-data.md
Comment (ghost) [reply docs/collections/returning-collection-data.md]: Thank you @jamesm-macrometa! 👍

**Comment by ghost** [comment]
Thanks, @diana-macrometa, and @jamesm-macrometa. Let's continue 🚀

**Comment by ghost** [comment]
What is the procedure after review? Can I merge to the main? 😅 @diana-macrometa

**Review by diana-macrometa — APPROVED** [review]
Good to go once you address James' suggestions.

**Pull request merged**
